be/src/olap/schema_change.cpp

```diff
@@ -1250,7 +1250,7 @@ OLAPStatus SchemaChangeHandler::_do_process_alter_tablet_v2(const TAlterTabletRe
                 rowsets_to_delete.push_back(rowset);
             }
         }
-        new_tablet->modify_rowsets(std::vector<RowsetSharedPtr>(), rowsets_to_delete);
+        new_tablet->delete_rowsets(rowsets_to_delete);
```

Comment:
I think there should be only one method calling delete rowsets. it is capture unused rowset.